### PR TITLE
Jenkins and dev environment bugfixes

### PR DESCRIPTION
**Pull request by Pooja-Gangwar**
### Desired Outcome

Jenkins builds were failing with the following logs when building the Ansible service:
```
E: Problem executing scripts APT::Update::Post-Invoke 'rm -f /var/cache/apt/archives/*.deb /var/cache/apt/archives/partial/*.deb /var/cache/apt/*.bin || true'
E: Sub-process returned an error code
```

This error occurs when running `apt-get` commands in Ubuntu 22.04 in Docker <= 20.10.9.
These Dockerfiles have been pinned to Ubuntu 20.04.

### Implemented Changes

- Dev env scripts now support `docker-compose` v1 and v2.
- Ubuntu-based Dockerfiles are pinned to 20.04.

### Connected Issue/Story

CyberArk internal issue link: [ONYX-19624](https://ca-il-jira.il.cyber-ark.com:8443/browse/ONYX-19624)

### Definition of Done
*At least 1 todo must be completed in the sections below for the PR to be
merged.*

#### Changelog

- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes, or
- [x] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]()
- [x] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [x] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [x] These changes are part of a larger initiative with a separate security review, or
- [ ] There are no security aspects to these changes

### CODE REVIEW

File: dev/playbooks/inventory-setup/inventory.j2 [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [testapp]
-{{ lookup('env','COMPOSE_PROJECT_NAME') }}-test_app_ubuntu-[1:2] ansible_connection=docker
-{{ lookup('env','COMPOSE_PROJECT_NAME') }}-test_app_centos-[1:2] ansible_connection=docker
+{{ lookup('env','COMPOSE_PROJECT_NAME') }}_test_app_ubuntu_[1:2] ansible_connection=docker
+{{ lookup('env','COMPOSE_PROJECT_NAME') }}_test_app_centos_[1:2] ansible_connection=docker
```
Comment (john-odonnell) [inline dev/playbooks/inventory-setup/inventory.j2:3]: Could you describe the bug you experiencing that these changes resolve? You've updated `inventory.j2` to use `docker-compose` v1 service naming scheme - it actually breaks the dev env for users with `docker-compose` v2.

I don't think we need to support the v1 naming scheme in the dev environment - I think we can expect up-to-date tools.

Comment (Pooja-Gangwar) [reply dev/playbooks/inventory-setup/inventory.j2:3]: No_Issue_AfterChanges
Issue_Without_Changes

Please find attached screen shots for reference , whole solution including role and dev works fine with these two lines changes .

Comment (Pooja-Gangwar) [reply dev/playbooks/inventory-setup/inventory.j2:3]: > Could you describe the bug you experiencing that these changes resolve? You've updated `inventory.j2` to use `docker-compose` v1 service naming scheme - it actually breaks the dev env for users with `docker-compose` v2.
> 
> I don't think we need to support the v1 naming scheme in the dev environment - I think we can expect up-to-date tools.

Thanks John , I have also added inventory-v2.j2 for docker-compose v2 .

**Review by john-odonnell — APPROVED** [review]
Rebased and approving, as this is blocking [ONYX-19910](https://ca-il-jira.il.cyber-ark.com:8443/browse/ONYX-19910). Thanks @Pooja-Gangwar

**Pull request merged**